src/man/sssd.conf.5.xml

```diff
@@ -1771,6 +1771,59 @@ pam_gssapi_services = sudo, sudo-i
                         </para>
                     </listitem>
                 </varlistentry>
+                <varlistentry>
+                    <term>pam_gssapi_indicators_map</term>
+                    <listitem>
+                        <para>
+                           Comma separated list of authentication indicators required
+                           to be present in a Kerberos ticket to access a PAM service
+                           that is allowed to try GSSAPI authentication using
+                           pam_sss_gss.so module.
+                        </para>
+                        <para>
+			   Each element of the list can be either authentication indicator
+                           name or a pair <quote>service:indicator</quote>. Indicators not
+			   prefixed with the PAM service name will be required to be present
+                           to access any PAM service configured to be used with
+                           <option>pam_gssapi_services</option>.
```

Comment:
It's not clear to me if items in the list of indicator are *match any* or *match all*. For example does ``pam_gssapi_indicators_map = sudo:pkinit, sudo:otp, hardened`` require either

* one or more indicators of `pkinit`, `otp`, and `hardened`
* all three indicators
* either `(pkinit or otp) and hardened`

Reply:
This option works together with the `pam_gssapi_services` which defines list of PAM services where `pam_sss_gss.so` would be allowed for use. `pam_gsssapi_indicators_map` provides a list of indicators that must be present when performing authentication for those PAM services. It is enough to have one of the listed indicators in the ticket. However, because people might want to have different settings for different PAM services, there is a way to set that per-PAM service with a prefix `service:`. To avoid repeating the same values, non-prefixed indicator applies to all services in `pam_gssapi_services`.

In your example, if `pam_gssapi_services = sudo` and `pam_gssapi_indicators_map = sudo:pkinit, sudo:otp, hardened`, when we'd do a check for `sudo` PAM service, then presence of either `pkinit`, `otp`, or `hardened` would allow to pass the check.

When `pam_sss_gss.so` applies authentication indicators check, the code will build a list of indicators that apply to the specific PAM service at the point of the check. Kerberos tickets typically have only one indicator, produced by the pre-authentication method used for obtaining TGT, so the code only checks that any of the indicators in the ticket match the list of indicators associated with the PAM service.

I don't think a presence of multiple indicators at the same time should be checked. I certainly don't see how we can express that in SSSD configuration without introducing a special parser. There might be no practical reason for that (yet?).

In practice, for IPA environments we only assign a single indicator during pre-authentication method processing. We folded up both FAST and SPAKE into the same `hardened` authentication indicator. We might add a recognition of MS-PKCA-obtained Kerberos tickets and inject `pkinit` indicator to service tickets in IPA for the users from Active Directory who used a smartcard to authenticate to their Windows machines. However, I don't see a need for the checks of multiple indicators at the same time.

Upstream, MIT Kerberos has internal tests that assign more than one indicator and verify that a received ticket has them all. So far it looks like a theoretical construct rather than a practical need. Being able to validate indicators against a list of values is more useful.

Reply:
Thanks for your clarification. I agree that an "match any" approach makes the most sense to me.

To me, the documentation of the ``pam_gssapi_indicators_map`` option does not make it obvious that it's a "match any" option.

Reply:
I'll update it.

Reply:
@tiran I extended man pages, hopefully, it is now more clear.